src/radical/pilot/pmgr/launching/base.py

```diff
@@ -562,38 +564,36 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
-        resource_manager        = rcfg.get('resource_manager')
-        pre_bootstrap_0         = rcfg.get('pre_bootstrap_0', [])
-        pre_bootstrap_1         = rcfg.get('pre_bootstrap_1', [])
-        python_interpreter      = rcfg.get('python_interpreter')
-        rp_version              = rcfg.get('rp_version')
-        virtenv_mode            = rcfg.get('virtenv_mode', DEFAULT_VIRTENV_MODE)
-        virtenv                 = rcfg.get('virtenv',      default_virtenv)
-        cores_per_node          = rcfg.get('cores_per_node', 0)
-        gpus_per_node           = rcfg.get('gpus_per_node',  0)
-        lfs_path_per_node       = rcfg.get('lfs_path_per_node')
-        lfs_size_per_node       = rcfg.get('lfs_size_per_node', 0)
-        python_dist             = rcfg.get('python_dist')
-        task_tmp                = rcfg.get('task_tmp')
-        spmd_variation          = rcfg.get('spmd_variation')
-        task_pre_launch         = rcfg.get('task_pre_launch')
-        task_pre_exec           = rcfg.get('task_pre_exec')
-        task_post_launch        = rcfg.get('task_post_launch')
-        task_post_exec          = rcfg.get('task_post_exec')
-        mandatory_args          = rcfg.get('mandatory_args', [])
-        system_architecture     = rcfg.get('system_architecture', {})
-        services               += rcfg.get('services', [])
+        agent_spawner           = rcfg.agent_spawner
+        agent_config            = rcfg.agent_config
+        agent_scheduler         = rcfg.agent_scheduler
+        default_queue           = rcfg.default_queue
+        forward_tunnel_endpoint = rcfg.forward_tunnel_endpoint
+        resource_manager        = rcfg.resource_manager
+        pre_bootstrap_0         = rcfg.pre_bootstrap_0     or list()
+        pre_bootstrap_1         = rcfg.pre_bootstrap_1     or list()
+        python_interpreter      = rcfg.python_interpreter
+        rp_version              = rcfg.rp_version
+        virtenv_mode            = rcfg.virtenv_mode
+        virtenv                 = rcfg.virtenv             or default_virtenv
+        cores_per_node          = rcfg.cores_per_node
+        gpus_per_node           = rcfg.gpus_per_node
+        lfs_path_per_node       = rcfg.lfs_path_per_node
+        lfs_size_per_node       = rcfg.lfs_size_per_node
+        python_dist             = rcfg.python_dist
+        task_tmp                = rcfg.task_tmp
+        task_pre_launch         = rcfg.task_pre_launch     or list()
+        task_post_launch        = rcfg.task_post_launch    or list()
+        task_pre_exec           = rcfg.task_pre_exec       or list()
+        task_post_exec          = rcfg.task_post_exec      or list()
+        mandatory_args          = rcfg.mandatory_args      or list()
+        system_architecture     = rcfg.system_architecture or dict()
+        services               += rcfg.services            or list()
+        raptor_cfg              = rcfg.raptor
```

Comment:
`or list()/dict()` looks excessive, since we set it with default values

Reply:
TODO: debug why defaults don't kick in...